docs/explanations/architecture/styles.md

```diff
@@ -0,0 +1,143 @@
+## Styles in the block editor
+
+This document introduces the main concepts related to styles in the block editor. It points to the relevant reference guides and tutorials for readers to dig deeper into each one of the ideas presented. It's aimed to block authors and people working in the block editor project.
+
+1. HTML and CSS
+2. User-provided Block Styles
+    - From UI controls to HTML markup
+    - Block Supports API
+    - Current limits of the Block Supports API
+3. User-provided Global Styles
+
+### 1. HTML and CSS
+
+By creating a post in the block editor the user is creating a number of artifacts: a HTML document plus a number of CSS stylesheets, either embedded in the document or external.
+
+On the one hand, the final HTML document is the result of a few things:
+
+- the [WordPress templates](https://developer.wordpress.org/themes/basics/template-files/) defined by the theme
```

Comment:
we now have two types of themes which might have two different behaviors related to styles, I wonder if that's worth mentioning at this point, maybe not.

Reply:
In terms of styles (themes with and without theme.json), I'd say we're covered ([see](https://github.com/WordPress/gutenberg/pull/38208/files#diff-ad9a82727f64bc4a98b87a8f43f18a72db4e6fed54f456a0f1b9fe082a0efa72R26)). I've added a note for people to dig into where the templates can come from in https://github.com/WordPress/gutenberg/pull/38208/commits/861f5eb145af17a13d8fc96327bc2548bc24a296